test/api/liskApi.js

```diff
@@ -695,4 +416,45 @@ describe('Lisk.api()', () => {
 			});
 		});
 	});
+
+
+	describe('#constructRequestData', () => {
+		const address = defaultAddress.address;
+		const optionsObject = {
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+		const expectedObject = {
+			address: '18160565574430594874L',
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+		const conflictObject = {
+			address: '18160565574430594874',
+			limit: 0,
+			offset: 5,
+		};
+		const resolvedConflictObject = {
+			address: '18160565574430594874',
+			limit: defaultRequestLimit,
+			offset: defaultRequestOffset,
+		};
+
+		it('should merge a data object with an options object', () => {
+			const createObject = privateApi.constructRequestData({ address }, optionsObject);
+			(createObject).should.be.eql(expectedObject);
+		});
+
+		it('should recognise when a callback function is passed instead of an options object', () => {
+			const createObject = privateApi.constructRequestData({ address }, () => true);
+			(createObject).should.be.eql({ address });
+		});
+
+		it('should prioritise values from the data object when the data object and options object conflict', () => {
+			const createObject = privateApi.constructRequestData(
+				{ limit: defaultRequestLimit, offset: defaultRequestOffset }, conflictObject,
+			);
```

Comment:
It would make more sense to me to have something like this:
```js
privateApi.constructRequestData(
    { address: '123L', limit: 4 }, // data
    { address, limit: defaultRequestLimit, offset: defaultRequestOffset }, // options
);
```
I.e. we have some options with a normal address and some defaults, and we want to override it with a weirdo address and an unusual limit, but keep whatever default options we haven't explicitly overridden.